### ISSUE TITLE
An option to retain and have access to whitespaces and comments

### ISSUE THREAD

**Issue by igordejanovic** [textX/textX#204]
Needed during transpiling, refactor, code augmentation etc.

See [here](https://github.com/textX/textX/issues/54#issuecomment-503527852)

Related to https://github.com/textX/Arpeggio/issues/56

**Comment by Aluriak** [comment]
Something like lossless syntax tree ? Many projects are using it, including [oil](http://www.oilshell.org/blog/2017/02/11.html) and the python package (red)[baron](https://github.com/PyCQA/baron).

**Comment by igordejanovic** [comment]
Exactly. The idea is to keep layout information. In the [hello world example](http://textx.github.io/textX/stable/tutorials/hello_world/) in point 3 it is referred to as `syntactic noise`. Those parts could be kept e.g. as a string on the next element in the model. E.g. keeping word `hello ` with  space after on the first `Who` object. This is just the basic idea. I expect a lot of corner cases here. :)

One very important use-cases for this feature is keeping DSL code up-to-date after meta-model/language change by automatic transformation. This is important as DSLs usually evolve rapidly.
